projectq/backends/_azure/_azure_quantum.py

```diff
@@ -0,0 +1,403 @@
+            return is_available_quantinuum(cmd)
+
+        return False
+
+    @property
+    def _target_factory(self):
+        target_factory = TargetFactory(
+            base_cls=Target,  # noqa
+            workspace=self._workspace,
+            default_targets=AzureQuantumBackend.DEFAULT_TARGETS
+        )
+
+        return target_factory
+
+    @property
+    def _target(self):
+        target = self._target_factory.get_targets(
+            name=self._target_name,
+            provider_id=self._provider_id
+        )
+
+        if type(target) is list and len(target) == 0:  # pragma: no cover
+            raise AzureQuantumTargetNotFoundError(
+                'Target {} is not available on workspace {}.'.format(
+                    self._target_name, self._workspace.name)
+            )
+
+        return target
+
+    @property
+    def current_availability(self):
+        """Current availability for given target."""
+        return self._target.current_availability
+
+    @property
+    def average_queue_time(self):
+        """Average queue time for given target."""
+        return self._target.average_queue_time
+
+    def get_probability(self, state, qureg):
+        """Shortcut to get a specific state's probability.
+
+        Args:
+            state (str): A state in bit-string format.
+            qureg (Qureg): A ProjectQ Qureg object.
+
+        Returns:
+            float: The probability for the provided state.
+        """
+        if len(state) != len(qureg):
+            raise ValueError('Desired state and register must be the same length!')
+
+        probs = self.get_probabilities(qureg)
+
+        try:
+            return probs[state]
+        except KeyError:
+            return 0.0
+
+    def get_probabilities(self, qureg):
+        """
+        Given the provided qubit register, determine the probability of each possible outcome.
+
+        Note:
+            This method should only be called *after* a circuit has been run and its results are available.
+
+        Args:
+            qureg (Qureg): A ProjectQ Qureg object.
+
+        Returns:
+            dict: A dict mapping of states -> probability.
+        """
+        if len(self._probabilities) == 0:
+            raise RuntimeError("Please, run the circuit first!")
+
+        probability_dict = {}
+        for state in self._probabilities:
+            mapped_state = ['0'] * len(qureg)
+            for i, qubit in enumerate(qureg):
+                try:
+                    meas_idx = self._measured_ids.index(qubit.id)
+                except ValueError:
+                    continue
+                mapped_state[i] = state[meas_idx]
+            probability = self._probabilities[state]
+            mapped_state = "".join(mapped_state)
+            probability_dict[mapped_state] = probability_dict.get(mapped_state, 0) + probability
+        return probability_dict
+
+    @property
+    def _input_data(self):
```

Comment:
Make sure you have consistent return statements in this function. Currently, there should not be a case where the control flow does not take either of the two `if` branches.

However, if some new provider gets added in the future, this function will most likely lead to some undefined behavior since there's an implicit `return None` at the end.

Either make this explicit with `return None` or raise an error at runtime.

Reply:
Make sense, since there are multiple of these scenarios, I find raising `RuntimeError` most consistent and appropriate.